src/compiler/checker.ts

```diff
@@ -7458,12 +7458,25 @@ namespace ts {
                     }
                 }
                 else {
-                    members.set(leftProp.name, leftProp);
+                    members.set(leftProp.name, getNonReadonlySymbol(leftProp));
                 }
             }
             return createAnonymousType(undefined, members, emptyArray, emptyArray, stringIndexInfo, numberIndexInfo);
         }
 
+        function getNonReadonlySymbol(prop: Symbol) {
+            if (!(getDeclarationModifierFlagsFromSymbol(prop) & ModifierFlags.Readonly)) {
+                return prop;
+            }
+            const declarations: Declaration[] = prop.declarations;
+            const flags = SymbolFlags.Property | (prop.flags & SymbolFlags.Optional);
+            const result = createSymbol(flags, prop.name);
+            result.checkFlags = 0;
+            result.type = getTypeOfSymbol(prop);
+            result.declarations = declarations;
+            return result;

```

Comment:
this will break find-all-refs unless `getRootSymbols` has a way to get back to the original symbol. Probably `mappedTypeOrigin` can be reused here.